ENIGMAsystem/SHELL/Graphics_Systems/Direct3D9/DX9surface.cpp

```diff
@@ -150,7 +146,6 @@ int surface_getpixel(int id, int x, int y)
   if (y < 0) y = 0;
   if (x > surface->width || y > surface->height) return 0;
 
-  d3dmgr->EndShapesBatching();
```

Comment:
I was not, hence it's back: 154375f5b36412d2ba17afb55444a4832dd391f4

Reason I did that was because I was on the thought process that reading the surface, while still rendering to it, was undefined. Now that I've put more thought into it, not defining that is problematic and definitely less compatible.

Side note, I want to clean up all of the duplicate surface code at some point. All they are is a composite of different kinds of textures and they can very easily be rebuilt on top of an abstraction. Another issue is that we don't necessarily want to flush unless you are requesting data from the currently bound surface (would be more efficient than what we're doing now, but this is still essentially a no-op if the batch isn't dirty anyway). We'll have to come back to this later.

Reply:
I should also mention, that I believe these surface changes are what made HitCoder's Sonic game start working in Direct3D9 on this pull request. So all around, I think we have definitely improved things here by making the batching more consistent across the systems.
![HitCoder's Sonic Game in Direct3D9](https://user-images.githubusercontent.com/3212801/44426231-13a5b000-a55c-11e8-9b88-345d88a2ad97.png)